satpy/composites/crefl_utils.py

```diff
@@ -331,6 +363,73 @@ def get_atm_variables(mus, muv, phi, height, coeffs):
     return sphalb, rhoray, TtotraytH2O, tOG
 
 
+def get_atm_variables_abi(mus, muv, phi, height, coeffs, G_O3, G_H2O, G_O2):
```

Comment:
there seems to be a lot of duplicated code in this function from existing function, can we factorise ?

Reply:
You are correct! I cooked something up and will push it later today or Monday (after I test to make sure I didn't break it).